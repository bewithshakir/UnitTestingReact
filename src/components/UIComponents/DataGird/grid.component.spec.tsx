import { shallow } from 'enzyme';
import GridComponent from "./grid.component";
import { OilCanIcon } from '../../../assets/icons';


describe('renders the Default Data Grid component', () => {
    const component = shallow(<GridComponent
        primaryKey={'taxJurisdictionId'}
        rows={[
            { col1: 'row-1-col-1', col2: 'row-1-col-2', col3: 'row-1-col-3', col4: 'row-1-col-4' },
            { col1: 'row-2-col-1', col2: 'row-2-col-2', col3: 'row-2-col-3', col4: 'row-2-col-4' },
            { col1: 'row-3-col-1', col2: 'row-3-col-2', col3: 'row-3-col-3', col4: 'row-3-col-4' },
            { col1: 'row-4-col-1', col2: 'row-4-col-2', col3: 'row-4-col-3', col4: 'row-4-col-4' },
        ]}
        header={[
            { field: "col1", label: "COL-1", type: 'text', align: 'left' },
            { field: "col2", label: "COL-2", type: 'text', align: 'left' },
            { field: "col3", label: "COL-3", type: 'text', align: 'left' },
            { field: "col4", label: "COL-4", type: 'text', align: 'left' },
        ]}
        isLoading={false}
        enableRowSelection={false}
        enableRowAction={false}
        getPages={jest.fn()}
        searchTerm=""
        openDrawer={jest.fn()}
        noDataMsg='Add New Data by clicking on the "Add" button.'
        getId={jest.fn()}
        resetCollaps={false}
        onResetTableCollaps={jest.fn()}
    />);

    it('snapshot test for Default Data Grid component', () => {
        expect(component).toMatchSnapshot();
    });

});

describe('renders Data Grid component with multiple accordian with child table', () => {
    const component = shallow(<GridComponent
        primaryKey={'taxJurisdictionId'}
        rows={[
            { col1: 'row-1-col-1', col2: 'row-1-col-2', col3: 'row-1-col-3', col4: 'row-1-col-4' },
            { col1: 'row-2-col-1', col2: 'row-2-col-2', col3: 'row-2-col-3', col4: 'row-2-col-4' },
            { col1: 'row-3-col-1', col2: 'row-3-col-2', col3: 'row-3-col-3', col4: 'row-3-col-4' },
            { col1: 'row-4-col-1', col2: 'row-4-col-2', col3: 'row-4-col-3', col4: 'row-4-col-4' },
        ]}
        header={[
            { field: "col1", label: "COL-1", type: 'text', align: 'left' },
            { field: "col2", label: "COL-2", type: 'button', align: 'left', icon: OilCanIcon, width: "350px" },
            { field: "col3", label: "COL-3", type: 'button', align: 'left', icon: OilCanIcon, width: "350px" },
            { field: "col4", label: "COL-4", type: 'text', align: 'left' },
        ]}
        isLoading={false}
        enableRowSelection={false}
        enableRowAction={false}
        getPages={jest.fn()}
        searchTerm=""
        openDrawer={jest.fn()}
        noDataMsg='Add New Data by clicking on the "Add" button.'
        getId={jest.fn()}
        resetCollaps={false}
        InnerTableComponent={{
            "COL-2": <GridComponent
                primaryKey={'taxJurisdictionId'}
                rows={[
                    { col1: 'sub-table-row-1-col-1', col2: 'sub-table-row-1-col-2', col3: 'sub-table-row-1-col-3', col4: 'sub-table-row-1-col-4' },
                    { col1: 'sub-table-row-2-col-1', col2: 'sub-table-row-2-col-2', col3: 'sub-table-row-2-col-3', col4: 'sub-table-row-2-col-4' },
                    { col1: 'sub-table-row-3-col-1', col2: 'sub-table-row-3-col-2', col3: 'sub-table-row-3-col-3', col4: 'sub-table-row-3-col-4' },
                    { col1: 'sub-table-row-4-col-1', col2: 'sub-table-row-4-col-2', col3: 'sub-table-row-4-col-3', col4: 'sub-table-row-4-col-4' },
                ]}
                header={[
                    { field: "col1", label: "Sub-Table-COL-1", type: 'text', align: 'left' },
                    { field: "col2", label: "Sub-Table-COL-2", type: 'text', align: 'left' },
                    { field: "col3", label: "Sub-Table-COL-3", type: 'text', align: 'left' },
                    { field: "col4", label: "Sub-Table-COL-4", type: 'text', align: 'left' },
                ]}
                isLoading={false}
                enableRowSelection={false}
                enableRowAction={false}
                getPages={jest.fn()}
                searchTerm=""
                openDrawer={jest.fn()}
                noDataMsg='Add New Data by clicking on the "Add" button.'
                getId={jest.fn()}
                resetCollaps={false}
                onResetTableCollaps={jest.fn()}
            />,
            "COL-3": <GridComponent
                primaryKey={'taxJurisdictionId'}
                rows={[
                    { col1: 'sub-table-row-1-col-1', col2: 'sub-table-row-1-col-2', col3: 'sub-table-row-1-col-3', col4: 'sub-table-row-1-col-4' },
                    { col1: 'sub-table-row-2-col-1', col2: 'sub-table-row-2-col-2', col3: 'sub-table-row-2-col-3', col4: 'sub-table-row-2-col-4' },
                    { col1: 'sub-table-row-3-col-1', col2: 'sub-table-row-3-col-2', col3: 'sub-table-row-3-col-3', col4: 'sub-table-row-3-col-4' },
                    { col1: 'sub-table-row-4-col-1', col2: 'sub-table-row-4-col-2', col3: 'sub-table-row-4-col-3', col4: 'sub-table-row-4-col-4' },
                ]}
                header={[
                    { field: "col1", label: "Sub-Table-COL-1", type: 'text', align: 'left' },
                    { field: "col2", label: "Sub-Table-COL-2", type: 'text', align: 'left' },
                    { field: "col3", label: "Sub-Table-COL-3", type: 'text', align: 'left' },
                    { field: "col4", label: "Sub-Table-COL-4", type: 'text', align: 'left' },
                ]}
                isLoading={false}
                enableRowSelection={false}
                enableRowAction={false}
                getPages={jest.fn()}
                searchTerm=""
                openDrawer={jest.fn()}
                noDataMsg='Add New Data by clicking on the "Add" button.'
                getId={jest.fn()}
                resetCollaps={false}
                onResetTableCollaps={jest.fn()}
            />
        }}
        onResetTableCollaps={jest.fn()}
    />);

    it('snapshot test for Default Data Grid component', () => {
        expect(component).toMatchSnapshot();
    });

});