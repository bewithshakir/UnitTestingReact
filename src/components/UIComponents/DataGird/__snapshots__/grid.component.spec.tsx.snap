// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Data Grid component with multiple accordian with child table snapshot test for Default Data Grid component 1`] = `
<ForwardRef(TableContainer)
  className="table-container"
  onScroll={[Function]}
>
  <ForwardRef(Table)
    aria-labelledby="tableTitle"
    stickyHeader={true}
    sx={
      Object {
        "minWidth": "1200px",
      }
    }
  >
    <EnhancedGridHead
      enableRowAction={false}
      enableRowSelection={false}
      headCells={
        Array [
          Object {
            "align": "left",
            "field": "col1",
            "label": "COL-1",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col2",
            "icon": [Function],
            "label": "COL-2",
            "type": "button",
            "width": "350px",
          },
          Object {
            "align": "left",
            "field": "col3",
            "icon": [Function],
            "label": "COL-3",
            "type": "button",
            "width": "350px",
          },
          Object {
            "align": "left",
            "field": "col4",
            "label": "COL-4",
            "type": "text",
          },
        ]
      }
      numSelected={0}
      onRequestSort={[Function]}
      onResetTableCollaps={[MockFunction]}
      onSelectAllClick={[Function]}
      order="asc"
      orderBy=""
      rowCount={4}
    />
    <EnhancedGridBody
      InnerTableComponent={
        Object {
          "COL-2": <GridComponent
            enableRowAction={false}
            enableRowSelection={false}
            getId={[MockFunction]}
            getPages={[MockFunction]}
            header={
              Array [
                Object {
                  "align": "left",
                  "field": "col1",
                  "label": "Sub-Table-COL-1",
                  "type": "text",
                },
                Object {
                  "align": "left",
                  "field": "col2",
                  "label": "Sub-Table-COL-2",
                  "type": "text",
                },
                Object {
                  "align": "left",
                  "field": "col3",
                  "label": "Sub-Table-COL-3",
                  "type": "text",
                },
                Object {
                  "align": "left",
                  "field": "col4",
                  "label": "Sub-Table-COL-4",
                  "type": "text",
                },
              ]
            }
            isLoading={false}
            noDataMsg="Add New Data by clicking on the \\"Add\\" button."
            onResetTableCollaps={[MockFunction]}
            openDrawer={[MockFunction]}
            primaryKey="taxJurisdictionId"
            resetCollaps={false}
            rowActionOptions={Array []}
            rows={
              Array [
                Object {
                  "col1": "sub-table-row-1-col-1",
                  "col2": "sub-table-row-1-col-2",
                  "col3": "sub-table-row-1-col-3",
                  "col4": "sub-table-row-1-col-4",
                },
                Object {
                  "col1": "sub-table-row-2-col-1",
                  "col2": "sub-table-row-2-col-2",
                  "col3": "sub-table-row-2-col-3",
                  "col4": "sub-table-row-2-col-4",
                },
                Object {
                  "col1": "sub-table-row-3-col-1",
                  "col2": "sub-table-row-3-col-2",
                  "col3": "sub-table-row-3-col-3",
                  "col4": "sub-table-row-3-col-4",
                },
                Object {
                  "col1": "sub-table-row-4-col-1",
                  "col2": "sub-table-row-4-col-2",
                  "col3": "sub-table-row-4-col-3",
                  "col4": "sub-table-row-4-col-4",
                },
              ]
            }
            searchTerm=""
          />,
          "COL-3": <GridComponent
            enableRowAction={false}
            enableRowSelection={false}
            getId={[MockFunction]}
            getPages={[MockFunction]}
            header={
              Array [
                Object {
                  "align": "left",
                  "field": "col1",
                  "label": "Sub-Table-COL-1",
                  "type": "text",
                },
                Object {
                  "align": "left",
                  "field": "col2",
                  "label": "Sub-Table-COL-2",
                  "type": "text",
                },
                Object {
                  "align": "left",
                  "field": "col3",
                  "label": "Sub-Table-COL-3",
                  "type": "text",
                },
                Object {
                  "align": "left",
                  "field": "col4",
                  "label": "Sub-Table-COL-4",
                  "type": "text",
                },
              ]
            }
            isLoading={false}
            noDataMsg="Add New Data by clicking on the \\"Add\\" button."
            onResetTableCollaps={[MockFunction]}
            openDrawer={[MockFunction]}
            primaryKey="taxJurisdictionId"
            resetCollaps={false}
            rowActionOptions={Array []}
            rows={
              Array [
                Object {
                  "col1": "sub-table-row-1-col-1",
                  "col2": "sub-table-row-1-col-2",
                  "col3": "sub-table-row-1-col-3",
                  "col4": "sub-table-row-1-col-4",
                },
                Object {
                  "col1": "sub-table-row-2-col-1",
                  "col2": "sub-table-row-2-col-2",
                  "col3": "sub-table-row-2-col-3",
                  "col4": "sub-table-row-2-col-4",
                },
                Object {
                  "col1": "sub-table-row-3-col-1",
                  "col2": "sub-table-row-3-col-2",
                  "col3": "sub-table-row-3-col-3",
                  "col4": "sub-table-row-3-col-4",
                },
                Object {
                  "col1": "sub-table-row-4-col-1",
                  "col2": "sub-table-row-4-col-2",
                  "col3": "sub-table-row-4-col-3",
                  "col4": "sub-table-row-4-col-4",
                },
              ]
            }
            searchTerm=""
          />,
        }
      }
      enableRowAction={false}
      enableRowSelection={false}
      getId={[MockFunction]}
      getPages={[MockFunction]}
      handleCheckChange={[Function]}
      headCells={
        Array [
          Object {
            "align": "left",
            "field": "col1",
            "label": "COL-1",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col2",
            "icon": [Function],
            "label": "COL-2",
            "type": "button",
            "width": "350px",
          },
          Object {
            "align": "left",
            "field": "col3",
            "icon": [Function],
            "label": "COL-3",
            "type": "button",
            "width": "350px",
          },
          Object {
            "align": "left",
            "field": "col4",
            "label": "COL-4",
            "type": "text",
          },
        ]
      }
      header={
        Array [
          Object {
            "align": "left",
            "field": "col1",
            "label": "COL-1",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col2",
            "icon": [Function],
            "label": "COL-2",
            "type": "button",
            "width": "350px",
          },
          Object {
            "align": "left",
            "field": "col3",
            "icon": [Function],
            "label": "COL-3",
            "type": "button",
            "width": "350px",
          },
          Object {
            "align": "left",
            "field": "col4",
            "label": "COL-4",
            "type": "text",
          },
        ]
      }
      isLoading={false}
      noDataMsg="Add New Data by clicking on the \\"Add\\" button."
      onResetTableCollaps={[MockFunction]}
      openDrawer={[MockFunction]}
      order="asc"
      orderBy=""
      primaryKey="taxJurisdictionId"
      resetCollaps={false}
      rowActionOptions={Array []}
      rows={
        Array [
          Object {
            "col1": "row-1-col-1",
            "col2": "row-1-col-2",
            "col3": "row-1-col-3",
            "col4": "row-1-col-4",
          },
          Object {
            "col1": "row-2-col-1",
            "col2": "row-2-col-2",
            "col3": "row-2-col-3",
            "col4": "row-2-col-4",
          },
          Object {
            "col1": "row-3-col-1",
            "col2": "row-3-col-2",
            "col3": "row-3-col-3",
            "col4": "row-3-col-4",
          },
          Object {
            "col1": "row-4-col-1",
            "col2": "row-4-col-2",
            "col3": "row-4-col-3",
            "col4": "row-4-col-4",
          },
        ]
      }
      searchTerm=""
      selectedRows={Array []}
    />
  </ForwardRef(Table)>
</ForwardRef(TableContainer)>
`;

exports[`renders the Default Data Grid component snapshot test for Default Data Grid component 1`] = `
<ForwardRef(TableContainer)
  className="table-container"
  onScroll={[Function]}
>
  <ForwardRef(Table)
    aria-labelledby="tableTitle"
    stickyHeader={true}
    sx={
      Object {
        "minWidth": "1200px",
      }
    }
  >
    <EnhancedGridHead
      enableRowAction={false}
      enableRowSelection={false}
      headCells={
        Array [
          Object {
            "align": "left",
            "field": "col1",
            "label": "COL-1",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col2",
            "label": "COL-2",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col3",
            "label": "COL-3",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col4",
            "label": "COL-4",
            "type": "text",
          },
        ]
      }
      numSelected={0}
      onRequestSort={[Function]}
      onResetTableCollaps={[MockFunction]}
      onSelectAllClick={[Function]}
      order="asc"
      orderBy=""
      rowCount={4}
    />
    <EnhancedGridBody
      enableRowAction={false}
      enableRowSelection={false}
      getId={[MockFunction]}
      getPages={[MockFunction]}
      handleCheckChange={[Function]}
      headCells={
        Array [
          Object {
            "align": "left",
            "field": "col1",
            "label": "COL-1",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col2",
            "label": "COL-2",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col3",
            "label": "COL-3",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col4",
            "label": "COL-4",
            "type": "text",
          },
        ]
      }
      header={
        Array [
          Object {
            "align": "left",
            "field": "col1",
            "label": "COL-1",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col2",
            "label": "COL-2",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col3",
            "label": "COL-3",
            "type": "text",
          },
          Object {
            "align": "left",
            "field": "col4",
            "label": "COL-4",
            "type": "text",
          },
        ]
      }
      isLoading={false}
      noDataMsg="Add New Data by clicking on the \\"Add\\" button."
      onResetTableCollaps={[MockFunction]}
      openDrawer={[MockFunction]}
      order="asc"
      orderBy=""
      primaryKey="taxJurisdictionId"
      resetCollaps={false}
      rowActionOptions={Array []}
      rows={
        Array [
          Object {
            "col1": "row-1-col-1",
            "col2": "row-1-col-2",
            "col3": "row-1-col-3",
            "col4": "row-1-col-4",
          },
          Object {
            "col1": "row-2-col-1",
            "col2": "row-2-col-2",
            "col3": "row-2-col-3",
            "col4": "row-2-col-4",
          },
          Object {
            "col1": "row-3-col-1",
            "col2": "row-3-col-2",
            "col3": "row-3-col-3",
            "col4": "row-3-col-4",
          },
          Object {
            "col1": "row-4-col-1",
            "col2": "row-4-col-2",
            "col3": "row-4-col-3",
            "col4": "row-4-col-4",
          },
        ]
      }
      searchTerm=""
      selectedRows={Array []}
    />
  </ForwardRef(Table)>
</ForwardRef(TableContainer)>
`;
